### PR TITLE
[WIP] MathOptInterface 0.6 support

### PR DESCRIPTION
**Pull request by GunnarFarneback**
Update for MathOptInterface 0.5. Only partially working and will need some hints on the way.

### CODE REVIEW

**Comment by simonschoelly** [comment]
I think line 179 should be changes to something like
`throw(MOI.UnsupportedConstraint{F,S}("Cbc MOI Interface does not support constraints of type " * (F,S) * "."))`
But then I'm absolutely no expert in this library - I just tried to get it to work a while ago but then go busy with other stuff.

**Comment by mlubin** [comment]
This is known to not work in MOI 0.5: https://github.com/JuliaOpt/MathOptInterface.jl/issues/463. Please update directly to MOI 0.6.

**Comment by GunnarFarneback** [comment]
Thanks for the link. Explains why this problem was difficult to get past.

File: REQUIRE [hunk 1]
```diff
@@ -1,7 +1,7 @@
 julia 0.6
 BinDeps
 MathProgBase 0.5.5 0.8
-MathOptInterface 0.4.1 0.5
+MathOptInterface 0.6
```
Comment (odow) [inline REQUIRE]: You should set an upper bound to avoid future breakages, so `MathOptInterface 0.6 0.7`

**Comment by GunnarFarneback** [comment]
Currently two errors in the tests run so far.

The first is again a problem with
```
    MOI.set(model, MOI.ConstraintSet(), c1, MOI.GreaterThan(100.0))
```
which goes into `replace_constraint_function_or_set` and fails to find a constraint in a map.

The second is a failure with
```
    @test MOI.supports_constraint(model, MOI.VectorAffineFunction{Float64}, MOI.Zeros)
```
I can add that combination to the `supports_constraint` method for `CbcOptimizer` but I don't know if it's the right thing to do and it's not sufficient anyway since the dispatch first goes to the `CachingOptimizer` method, which also requires `MOI.supports_constraint(m.model_cache, F, S)`, where I'm currently lost.

Any insights into these errors would be welcome.

**Comment by vitornesello** [comment]
@GunnarFarneback the cause of the first error is that the constraint indices for SingleVariable constraints are not added to the map right now, but they should be added. Thats why he does not find the key in the map. Check these lines:
```julia
         else
             ## Update conmap for (F,S) for F != MOI.SingleVariable
             ## Single variables are treated by bounds in Cbc, so no
             ## need to add a row
             for i in 1:length(ci)
                 mapping.conmap[ci[i]] = MOI.ConstraintIndex{F,S}(num_rows + i)
             end
             num_rows += MOI.get(user_optimizer, MOI.NumberOfConstraints{F,S}())
         end
```

**Comment by GunnarFarneback** [comment]
Thanks for the hint but I'm afraid fixing this is beyond my familiarity with the code base and the MOI design ideas. I could maybe work through it with enough hand-holding, but it's probably more efficient if someone who actually knows what they're doing take over.

**Comment by mlubin** [comment]
Closed by #68

**Pull request closed without merging**